src/main/java/internet/market/security/AuthenticationServiceImpl.java

```diff
@@ -16,7 +17,7 @@ public User login(String login, String password) throws AuthenticationException
         User user = userService.findByLogin(login)
                 .orElseThrow(() ->
                         new AuthenticationException("Wrong password or login"));
-        if (user.getPassword().equals(password)) {
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
             return user;
         }
         throw new AuthenticationException("Wrong password or login");
```

Comment:
can we modify it with isPresent() and throw exception only in one place?

Reply:
done